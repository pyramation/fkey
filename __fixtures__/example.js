export default {
  data: {
    _meta: {
      tables: [
        {
          name: 'ActionGoal',
          foreignKeyConstraints: [
            {
              refTable: {
                name: 'Action'
              }
            },
            {
              refTable: {
                name: 'Goal'
              }
            },
            {
              refTable: {
                name: 'User'
              }
            }
          ]
        },
        {
          name: 'ActionItem',
          foreignKeyConstraints: [
            {
              refTable: {
                name: 'Action'
              }
            },
            {
              refTable: {
                name: 'User'
              }
            }
          ]
        },
        {
          name: 'ActionResult',
          foreignKeyConstraints: [
            {
              refTable: {
                name: 'Action'
              }
            },
            {
              refTable: {
                name: 'User'
              }
            }
          ]
        },
        {
          name: 'Action',
          foreignKeyConstraints: [
            {
              refTable: {
                name: 'User'
              }
            }
          ]
        },
        {
          name: 'GoalExplanation',
          foreignKeyConstraints: [
            {
              refTable: {
                name: 'Goal'
              }
            }
          ]
        },
        {
          name: 'Goal',
          foreignKeyConstraints: []
        },
        {
          name: 'NewsUpdate',
          foreignKeyConstraints: []
        },
        {
          name: 'OrganizationProfile',
          foreignKeyConstraints: [
            {
              refTable: {
                name: 'User'
              }
            }
          ]
        },
        {
          name: 'UserActionItem',
          foreignKeyConstraints: [
            {
              refTable: {
                name: 'User'
              }
            },
            {
              refTable: {
                name: 'Action'
              }
            },
            {
              refTable: {
                name: 'UserAction'
              }
            },
            {
              refTable: {
                name: 'ActionItem'
              }
            }
          ]
        },
        {
          name: 'UserActionResult',
          foreignKeyConstraints: [
            {
              refTable: {
                name: 'User'
              }
            },
            {
              refTable: {
                name: 'Action'
              }
            },
            {
              refTable: {
                name: 'UserAction'
              }
            },
            {
              refTable: {
                name: 'ActionResult'
              }
            }
          ]
        },
        {
          name: 'UserAction',
          foreignKeyConstraints: [
            {
              refTable: {
                name: 'User'
              }
            },
            {
              refTable: {
                name: 'User'
              }
            },
            {
              refTable: {
                name: 'Action'
              }
            }
          ]
        },
        {
          name: 'UserCharacteristic',
          foreignKeyConstraints: [
            {
              refTable: {
                name: 'User'
              }
            }
          ]
        },
        {
          name: 'UserConnection',
          foreignKeyConstraints: [
            {
              refTable: {
                name: 'User'
              }
            },
            {
              refTable: {
                name: 'User'
              }
            }
          ]
        },
        {
          name: 'UserContact',
          foreignKeyConstraints: [
            {
              refTable: {
                name: 'User'
              }
            }
          ]
        },
        {
          name: 'UserEmail',
          foreignKeyConstraints: [
            {
              refTable: {
                name: 'User'
              }
            }
          ]
        },
        {
          name: 'UserPassAction',
          foreignKeyConstraints: [
            {
              refTable: {
                name: 'User'
              }
            },
            {
              refTable: {
                name: 'Action'
              }
            }
          ]
        },
        {
          name: 'UserProfile',
          foreignKeyConstraints: [
            {
              refTable: {
                name: 'User'
              }
            }
          ]
        },
        {
          name: 'UserSavedAction',
          foreignKeyConstraints: [
            {
              refTable: {
                name: 'User'
              }
            },
            {
              refTable: {
                name: 'Action'
              }
            }
          ]
        },
        {
          name: 'UserSetting',
          foreignKeyConstraints: [
            {
              refTable: {
                name: 'User'
              }
            }
          ]
        },
        {
          name: 'User',
          foreignKeyConstraints: []
        },
        {
          name: 'ZipCode',
          foreignKeyConstraints: []
        }
      ]
    }
  }
};
