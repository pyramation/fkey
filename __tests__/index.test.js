import example from '../__fixtures__/example';
import parse from '../src';
const meta = example.data._meta;

it('works', async () => {
  expect(parse(meta)).toMatchSnapshot();
});
