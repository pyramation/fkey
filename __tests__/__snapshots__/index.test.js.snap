// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`works 1`] = `
Object {
  "all": Object {
    "Action": Array [
      "User",
      "Action",
    ],
    "ActionGoal": Array [
      "User",
      "Action",
      "Goal",
      "ActionGoal",
    ],
    "ActionItem": Array [
      "User",
      "Action",
      "ActionItem",
    ],
    "ActionResult": Array [
      "User",
      "Action",
      "ActionResult",
    ],
    "EVERYTHING_INCLUDED": Array [
      "User",
      "Action",
      "Goal",
      "ActionGoal",
      "ActionItem",
      "ActionResult",
      "GoalExplanation",
      "NewsUpdate",
      "OrganizationProfile",
      "UserAction",
      "UserActionItem",
      "UserActionResult",
      "UserCharacteristic",
      "UserConnection",
      "UserContact",
      "UserEmail",
      "UserPassAction",
      "UserProfile",
      "UserSavedAction",
      "UserSetting",
      "ZipCode",
      "EVERYTHING_INCLUDED",
    ],
    "Goal": Array [
      "Goal",
    ],
    "GoalExplanation": Array [
      "Goal",
      "GoalExplanation",
    ],
    "NewsUpdate": Array [
      "NewsUpdate",
    ],
    "OrganizationProfile": Array [
      "User",
      "OrganizationProfile",
    ],
    "User": Array [
      "User",
    ],
    "UserAction": Array [
      "User",
      "Action",
      "UserAction",
    ],
    "UserActionItem": Array [
      "User",
      "Action",
      "UserAction",
      "ActionItem",
      "UserActionItem",
    ],
    "UserActionResult": Array [
      "User",
      "Action",
      "UserAction",
      "ActionResult",
      "UserActionResult",
    ],
    "UserCharacteristic": Array [
      "User",
      "UserCharacteristic",
    ],
    "UserConnection": Array [
      "User",
      "UserConnection",
    ],
    "UserContact": Array [
      "User",
      "UserContact",
    ],
    "UserEmail": Array [
      "User",
      "UserEmail",
    ],
    "UserPassAction": Array [
      "User",
      "Action",
      "UserPassAction",
    ],
    "UserProfile": Array [
      "User",
      "UserProfile",
    ],
    "UserSavedAction": Array [
      "User",
      "Action",
      "UserSavedAction",
    ],
    "UserSetting": Array [
      "User",
      "UserSetting",
    ],
    "ZipCode": Array [
      "ZipCode",
    ],
  },
  "longest": Array [
    "User",
    "Action",
    "Goal",
    "ActionGoal",
    "ActionItem",
    "ActionResult",
    "GoalExplanation",
    "NewsUpdate",
    "OrganizationProfile",
    "UserAction",
    "UserActionItem",
    "UserActionResult",
    "UserCharacteristic",
    "UserConnection",
    "UserContact",
    "UserEmail",
    "UserPassAction",
    "UserProfile",
    "UserSavedAction",
    "UserSetting",
    "ZipCode",
    "EVERYTHING_INCLUDED",
  ],
}
`;
